### PR TITLE
Follow godot linter recommendations

### PR DESCRIPTION
**Pull request by nachtmaar**
**Description**

Changes proposed in this pull request:

- Fix the lint warnings coming from godot (introduced by [this PR](https://github.com/kyma-project/kyma/pull/15383))

This also affects our CRD schema (as it is generated from the go code).
I checked ` k explain pod.spec` and it also uses trailings dots :)

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/pkg/backend/metrics/collector.go [hunk 1]
```diff
@@ -8,26 +8,26 @@ import (
...
 
-// Collector implements the prometheus.Collector interface
+// Collector implements the prometheus.Collector interface.
 type Collector struct {
 	deliveryPerSubscription *prometheus.CounterVec
 	eventTypes              *prometheus.CounterVec
 }
 
-// NewCollector a new instance of Collector
+// NewCollector a new instance of Collector.
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/metrics/collector.go:30]: this sentence no verb :)

File: components/eventing-controller/pkg/backend/nats/core/core.go [hunk 2]
```diff
@@ -232,12 +232,12 @@ func (n *Nats) GetInvalidSubscriptions() *[]types.NamespacedName {
 }
 
 // GetAllSubscriptions returns the map which contains all details of subscription
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core.go:234]:
```suggestion
// GetAllSubscriptions returns the map which contains all details of subscription.
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 3]
```diff
@@ -178,7 +178,7 @@ func TestSubscription(t *testing.T) {
 
 // TestNatsSubAfterSync_NoChange tests the SyncSubscription method
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:179]:
```suggestion
// TestNatsSubAfterSync_NoChange tests the SyncSubscription method.
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 4]
```diff
@@ -178,7 +178,7 @@ func TestSubscription(t *testing.T) {
...
 // when there is no change in the subscription then the method should
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:180]:
```suggestion
// When there is no change in the subscription then the method should
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 5]
```diff
@@ -261,7 +261,7 @@ func TestNatsSubAfterSync_NoChange(t *testing.T) {
 
 // TestNatsSubAfterSync_SinkChange tests the SyncSubscription method
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:262]:
```suggestion
// TestNatsSubAfterSync_SinkChange tests the SyncSubscription method.
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 6]
```diff
@@ -261,7 +261,7 @@ func TestNatsSubAfterSync_NoChange(t *testing.T) {
...
 // when only the sink is changed in subscription, then it should not re-create
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:263]:
```suggestion
// When only the sink is changed in subscription, then it should not re-create
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 7]
```diff
@@ -640,7 +640,7 @@ func TestNatsSubAfterSync_FilterRemoved(t *testing.T) {
 
 // TestNatsSubAfterSync_MultipleSubs tests the SyncSubscription method
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:641]:
```suggestion
// TestNatsSubAfterSync_MultipleSubs tests the SyncSubscription method.
```

File: components/eventing-controller/pkg/backend/nats/core/core_test.go [hunk 8]
```diff
@@ -640,7 +640,7 @@ func TestNatsSubAfterSync_FilterRemoved(t *testing.T) {
...
 // when there are two subscriptions and the filter is changed in one subscription
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/nats/core/core_test.go:642]:
```suggestion
// When there are two subscriptions and the filter is changed in one subscription
```

File: components/eventing-controller/pkg/backend/utils/utils.go [hunk 9]
```diff
@@ -60,7 +60,7 @@ func hashSubscriptionFullName(domainName, namespace, name string) string {
 }
 
 // produces a name+hash which is not longer than maxLength. If necessary, shortens name, not the hash.
```
Comment (k15r) [inline components/eventing-controller/pkg/backend/utils/utils.go:62]:
```suggestion
// shortenNameAndAppendHash produces a `name+hash` which is not longer than maxLength. If necessary, shortens name, not the hash.
```

File: components/eventing-controller/pkg/object/apirule.go [hunk 10]
```diff
@@ -9,7 +9,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// OAuthHandlerName OAuth handler name supported in Kyma
+// OAuthHandlerName OAuth handler name supported in Kyma.
```
Comment (k15r) [inline components/eventing-controller/pkg/object/apirule.go:12]: ???

File: components/eventing-controller/testing/bebmock.go [hunk 11]
```diff
@@ -220,8 +220,8 @@ func IsBEBSubscriptionDelete(r *http.Request) bool {
 	return r.Method == http.MethodDelete && strings.Contains(r.RequestURI, MessagingURLPath)
 }
 
-// GetRestAPIObject gets the name of the involved object in a REST url
-// e.g. "/messaging/events/subscriptions/{subscriptionName}" => "{subscriptionName}"
+// GetRestAPIObject gets the name of the involved object in a REST url.
```
Comment (k15r) [inline components/eventing-controller/testing/bebmock.go:223]:
```suggestion
// GetRestAPIObject gets the name of the involved object in a REST url,
```

File: components/eventing-controller/testing/test_helpers.go [hunk 12]
```diff
@@ -383,7 +383,7 @@ func WithFilter(eventSource, eventType string) SubscriptionOpt {
 }
 
 // WithNotCleanFilter initializes subscription filter with a not clean event-type
```
Comment (k15r) [inline components/eventing-controller/testing/test_helpers.go:385]:
```suggestion
// WithNotCleanFilter initializes subscription filter with a not clean event-type.
```

**Review by k15r — APPROVED** [review]

**Comment by nachtmaar** [comment]
@k15r Thank you for approving the PR. As discussed I will work on your suggestions as soon as I am back from my vacation :)

**Pull request merged**